Try/catch property deprecation warnings to support IE8

syranide opened the pull request:
Tested separately and against our app. Starts and works correctly now.
I removed the if-condition simply because wasn't used else-where and seems redundant for DEV.

cc @sebmarkbage

zpao commented:
In the future it would help if you explained *why* this makes it so IE8 works, not just that it does :)

For reference: it's because `Object.defineProperty` does exist but only works for DOM node objects in IE8 and throws otherwise. Using the shim/sham may or may not help.

Side note, this change does align with how we're using `defineProperty` in `ReactElement`.

syranide commented:
@zpao I always try to, but I figured since this was already used elsewhere (for the same reason) that it was implied and I just added it to the new locations that was missing them (I even copied the comment and just added es5-shim/sham for posterity). :)

:+1:
